beacon_node/beacon_chain/src/fetch_blobs.rs

```diff
@@ -0,0 +1,333 @@
+            )
+            .discard_timer_on_break(&mut timer);
+            drop(timer);
+
+            let mut all_data_columns = match data_columns_result {
+                Ok(d) => d,
+                Err(e) => {
+                    error!(
+                        log,
+                        "Failed to build data column sidecars from blobs";
+                        "error" => ?e
+                    );
+                    return;
+                }
+            };
+
+            if let Err(e) = data_columns_sender.try_send(all_data_columns.clone()) {
+                error!(log, "Failed to send computed data columns"; "error" => ?e);
+            };
+
+            // Check indices from cache before sending the columns, to make sure we don't
+            // publish components already seen on gossip.
+            let is_supernode = chain_cloned
+                .data_availability_checker
+                .get_sampling_column_count()
+                == chain_cloned.spec.number_of_columns;
+
+            // At the moment non supernodes are not required to publish any columns.
+            // TODO(das): we could experiment with having full nodes publish their custodied
+            // columns here.
+            if !is_supernode {
+                return;
+            }
+
+            // To reduce bandwidth for supernodes: permute the columns to publish and
+            // publish them in batches. Our hope is that some columns arrive from
+            // other supernodes in the meantime, obviating the need for us to publish
+            // them. If no other publisher exists for a column, it will eventually get
+            // published here.
+            // FIXME(das): deduplicate this wrt to gossip/sync methods
+            all_data_columns.shuffle(&mut rand::thread_rng());
+
+            let supernode_data_column_publication_batches = chain_cloned
+                .config
+                .supernode_data_column_publication_batches;
+            let supernode_data_column_publication_batch_interval = chain_cloned
+                .config
+                .supernode_data_column_publication_batch_interval;
+
+            let batch_size = all_data_columns.len() / supernode_data_column_publication_batches;
+            for batch in all_data_columns.chunks(batch_size) {
+                let already_seen = chain_cloned
+                    .data_availability_checker
+                    .cached_data_column_indexes(&block_root)
+                    .unwrap_or_default();
+                let publishable = batch
+                    .iter()
+                    .filter(|col| !already_seen.contains(&col.index()))
+                    .cloned()
+                    .collect::<Vec<_>>();
+                if !publishable.is_empty() {
+                    debug!(
+                        log,
+                        "Publishing data columns from EL";
+                        "count" => publishable.len()
+                    );
```

Comment:
Added more logging in 81493f5e67da03aabbffbbf3c0d11f754e27ec21

> Also, do we log somewhere which specific column index we publish? Otherwise, it might be helpful to log in this line a vec of indices of which columns we are publishing

Yes this is logged in `network_service`:
https://github.com/sigp/lighthouse/blob/9aefb5539baff637d68deb3dd386ff45312f3573/beacon_node/network/src/service.rs#L666-L671